src/Symfony/Component/Finder/Glob.php

```diff
@@ -54,16 +54,18 @@ public static function toRegex($glob, $strictLeadingDot = true, $strictWildcardS
         $sizeGlob = strlen($glob);
         for ($i = 0; $i < $sizeGlob; ++$i) {
             $car = $glob[$i];
-            if ($firstByte) {
-                if ($strictLeadingDot && '.' !== $car) {
-                    $regex .= '(?=[^\.])';
-                }
-
-                $firstByte = false;
+            if ($firstByte && $strictLeadingDot && '.' !== $car) {
+                $regex .= '(?=[^\.])';
             }
 
-            if ('/' === $car) {
-                $firstByte = true;
+            $firstByte = '/' === $car;
+
+            if ($firstByte && $strictWildcardSlash && isset($glob[$i + 3]) && '**/' === $glob[$i + 1].$glob[$i + 2].$glob[$i + 3]) {
+                $car = $strictLeadingDot ? '/((?=[^\.])[^/]+/)*' : '/([^/]+/)*';
```

Comment:
I would use a possessive quantifier for ``[^/]++/`` to avoid useless backtracking.